streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/AssignorConfigurationTest.java

```diff
@@ -54,30 +54,25 @@ public void configsShouldRejectZeroWarmups() {
     }
 
     @Test
-    public void rebalanceProtocolShouldSupportAllUpgradeFromVersions() {
+    public void shouldSupportAllUpgradeFromVersionsFromCooperativeRebalancingOn() {
+        boolean beforeCooperative = true;
         for (final UpgradeFromValues upgradeFrom : UpgradeFromValues.values()) {
-            config.put(StreamsConfig.UPGRADE_FROM_CONFIG, upgradeFrom.toString());
-            final AssignorConfiguration assignorConfiguration = new AssignorConfiguration(config);
-
-            try {
-                assignorConfiguration.rebalanceProtocol();
-            } catch (final Exception error) {
-                throw new AssertionError("Upgrade from " + upgradeFrom + " failed with " + error.getMessage() + "!");
+            if (upgradeFrom.toString().equals("2.4")) {
+                beforeCooperative = false;
             }
-        }
-    }
 
-    @Test
-    public void configuredMetadataVersionShouldSupportAllUpgradeFromVersions() {
-        for (final UpgradeFromValues upgradeFrom : UpgradeFromValues.values()) {
             config.put(StreamsConfig.UPGRADE_FROM_CONFIG, upgradeFrom.toString());
-            final AssignorConfiguration assignorConfiguration = new AssignorConfiguration(config);
 
-            try {
-                assignorConfiguration.configuredMetadataVersion(0);
-            } catch (final Exception error) {
-                throw new AssertionError("Upgrade from " + upgradeFrom + " failed with " + error.getMessage() + "!");
+            if (beforeCooperative) {
+                assertThrows(IllegalStateException.class, () -> new AssignorConfiguration(config));
+            } else {
+                try {
+                    final AssignorConfiguration assignorConfiguration = new AssignorConfiguration(config);
+                } catch (final Exception error) {
+                    throw new AssertionError("Upgrade from " + upgradeFrom + " failed with " + error.getMessage() + "!");
```

Comment:
Why do we catch and throw `AssertionError`? It seems we can just remove the try-catch as we don't expect any exception, and if we hit an exception the test fails anyway?

If you want to make it explicit you can use `assertDoesNotThrow`.

Reply:
don't blame me this is just what the test was doing before (unless I wrote that original test in which case you can still blame me 😝 )

Reply:
_disappearing_